android-sdk/src/androidTest/java/com/example/android_sdk/PaymentFormPresentedEventTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.android_sdk
+
+import android.content.Context
+import android.os.Build
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class PaymentFormPresentedEventTest {
+    private var appContext: Context? = null
+
+    @Before
+    fun setUp() {
+        appContext = InstrumentationRegistry.getInstrumentation().targetContext
+    }
+
+    @Test
+    fun test_language_property_of_paymentFormPresented_event_returnsCorrectValue() {
+        setLocale(Locale.GERMANY)
+        assertTrue("de" == Locale.getDefault().language)
+    }
+
+    @Test
+    fun test_language_property_of_paymentFormPresented_event_returnWrongValue() {
+        setLocale(Locale.FRENCH)
+        assertFalse("en" == Locale.getDefault().language)
+    }
```

Comment:
what is the purpose of this test?

Reply:
This test is asserting the failure when the user changed the device language and in the worst case if the old configuration is not updated for the application.